File: myskills/backend/app/routers/agent.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db

router = APIRouter()

@router.post("/select-knowledge-bases", response_model=dict)
async def select_knowledge_bases(query_data: dict, db: Session = Depends(get_db)):
    """选择知识库（智能体推荐）"""
    # 这里只是一个占位实现，实际推荐逻辑需要在服务层实现
    return {
        "recommended_knowledge_bases": [
            {
                "id": "1",
                "name": "技术文档",
                "score": 0.95
            },
            {
                "id": "2",
                "name": "产品文档",
                "score": 0.85
            }
        ]
    }

File: myskills/frontend/src/services/agent.ts
import api from './api';

interface RecommendedKnowledgeBase {
  id: string;
  name: string;
  score: number;
}

interface AgentTemplate {
  id: string;
  name: string;
  type: string;
  description: string;
  status: string;
}

interface AgentTemplateDetail extends AgentTemplate {
  templates: any[];
}

class AgentService {
  async selectKnowledgeBases(query: string): Promise<{
    recommended_knowledge_bases: RecommendedKnowledgeBase[];
  }> {
    return api.post('/agent/select-knowledge-bases', { query });
  }

  async getAgentTemplates(): Promise<AgentTemplate[]> {
    return api.get('/agent-templates');
  }

  async getAgentTemplate(id: string): Promise<AgentTemplateDetail> {
    return api.get(`/agent-templates/${id}`);
  }
}

export default new AgentService();
